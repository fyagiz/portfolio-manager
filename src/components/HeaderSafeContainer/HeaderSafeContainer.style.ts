import { Platform, StyleSheet } from "react-native";
import { StatusBar } from "react-native";

export default StyleSheet.create({
  container: {
    flex: 1,
    marginTop: Platform.OS === "android" ? StatusBar.currentHeight : 0,
    alignContent: "center",
    justifyContent: "center",
    backgroundColor: "#017387",
    padding: 4,
  },
});
