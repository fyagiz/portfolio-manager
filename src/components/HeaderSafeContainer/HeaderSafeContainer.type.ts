import { ReactNode } from "react";

export type HeaderSafeContainerPropsType = {
  children: ReactNode;
};
