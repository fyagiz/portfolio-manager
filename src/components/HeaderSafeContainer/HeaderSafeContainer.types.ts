import { ReactNode } from "react";
import { ViewProps } from "react-native";

export type HeaderSafeContainerProps = {
  children: ReactNode;
  safeAreaViewProps?: ViewProps;
};
