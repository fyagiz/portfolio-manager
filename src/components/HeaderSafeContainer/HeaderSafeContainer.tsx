import { SafeAreaProvider } from "react-native-safe-area-context";
import { HeaderSafeContainerPropsType } from "./HeaderSafeContainer.type";

const HeaderSafeContainer = (props: HeaderSafeContainerPropsType) => {
  const { children } = props;
  return <SafeAreaProvider>{children}</SafeAreaProvider>;
};

export default HeaderSafeContainer;
