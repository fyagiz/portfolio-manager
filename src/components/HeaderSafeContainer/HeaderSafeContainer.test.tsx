import { render } from "@testing-library/react-native";
import HeaderSafeContainer from "./HeaderSafeContainer";
import { Button } from "react-native";

/*
1- HeaderSafeContainer should be rendered.
2- HeaderSafeContainer should render childrens.
*/

describe("HeaderSafe Component Unit Tests", () => {
  test("should render correctly", () => {
    const headerSafeContainerComponent = render(
      <HeaderSafeContainer>
        <Button title="TestButton" />
      </HeaderSafeContainer>,
    );
    expect(headerSafeContainerComponent).toMatchSnapshot();
  });
  test("should display children component", () => {
    const headerSafeContainerComponent = render(
      <HeaderSafeContainer>
        <Button title="TestButton" testID="testButton" />
      </HeaderSafeContainer>,
    );
    const testButton = headerSafeContainerComponent.getByTestId("testButton");
    expect(testButton).not.toBeNull();
  });
});
