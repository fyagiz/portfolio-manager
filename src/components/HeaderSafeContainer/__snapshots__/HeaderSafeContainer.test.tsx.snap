// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSafe Component Unit Tests should render correctly 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "alignContent": "center",
        "backgroundColor": "#017387",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 0,
        "padding": 4,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        TestButton
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
